### PR TITLE
feat: DM Push Notifications and Native iOS Profile Updates 

### PR DESCRIPTION
**Pull request by alexrisch**
## Release Notes

- **New Features**
  - Introduced functionality for managing account-specific notification subscriptions.
  - Added new methods for retrieving profiles based on account and inbox ID.
  - Implemented a new observer for managing conversation list queries.

- **Improvements**
  - Enhanced logging for notification subscriptions and profile retrieval processes.
  - Streamlined spam score computation logic.
  - Improved data handling and caching mechanisms for profile queries.

- **Refactor**
  - Updated conversation fetching methods to improve efficiency and clarity.
  - Refined notification handling logic for better readability and maintainability.

- **Bug Fixes**
  - Improved error handling in various notification and profile management functions.

### CODE REVIEW

File: queries/useInboxProfileSocialsQuery.ts [hunk 1]
```diff
@@ -9,14 +9,17 @@ import {
...
 import { InboxId } from "@xmtp/react-native-sdk";
-import { reactQueryPersister } from "@/utils/mmkv";
+import mmkv from "@/utils/mmkv";
 
 const profileSocialsQueryKey = (account: string, peerAddress: string) => [
   "inboxProfileSocials",
-  account,
-  peerAddress,
+  account?.toLowerCase(),
+  peerAddress?.toLowerCase(),
```
Comment (thierryskoda) [inline queries/useInboxProfileSocialsQuery.ts:17]: Shouldn't need `?` but nooo problem

File: queries/useProfileSocialsQuery.ts [hunk 2]
```diff
@@ -8,16 +8,19 @@ import {
...
 import { queryClient } from "./queryClient";
-import { reactQueryPersister } from "@/utils/mmkv";
+import mmkv from "@/utils/mmkv";
 
 type ProfileSocialsData = ProfileSocials | null | undefined;
 
 const profileSocialsQueryKey = (account: string, peerAddress: string) => [
   "profileSocials",
-  account,
+  account?.toLowerCase(),
```
Comment (thierryskoda) [inline queries/useProfileSocialsQuery.ts:17]: No `?` needed

File: queries/useV3ConversationListQuery.ts [hunk 3]
```diff
@@ -91,6 +94,17 @@ const v3ConversationListQueryConfig = (
   enabled: !!account,
 });
 
+export const createV3ConversationListQueryObserver = (
+  account: string,
+  context: string,
+  includeSync: boolean = true
+) => {
```
Comment (thierryskoda) [inline queries/useV3ConversationListQuery.ts:101]: when we have 3+ args I think it's better to do object args. I do even when I have 2

File: utils/mmkv.ts [hunk 4]
```diff
@@ -59,6 +59,7 @@ const reactQuerySyncStorage = {
     return stringValue || null;
   },
   setItem: (key: string, value: string) => {
+    console.log("setItem111", key, value);
```
Comment (thierryskoda) [inline utils/mmkv.ts:62]: Log

File: utils/xmtpRN/sync.ts [hunk 5]
```diff
@@ -128,8 +128,8 @@ const syncClientConversationList = async (account: string) => {
     });
     // Prefetch the conversation list so when we land on the conversation list
     // we have it ready, this will include syncing all groups
-    const conversationList = await fetchConversationListQuery(account);
-    subscribeToNotifications({ conversations: conversationList, account });
+    setupAccountTopicSubscription(account);
+    await prefetchConversationListQuery(account);
```
Comment (thierryskoda) [inline utils/xmtpRN/sync.ts:132]: Super good for now, just wonder if later we'll centralize this else where. I feel having this in `xmtpRN/sync.ts` isn't good? 🤔 

Maybe once we refactor the whole account store setup

**Review by thierryskoda — APPROVED** [review]
Let's try to name files (especially new ones) with kebab case 🙏

File: utils/xmtpRN/sync.ts (hunk 5 above)
Comment (alexrisch) [reply utils/xmtpRN/sync.ts:132]: Yeah I definitely this this will be removed, but for now want it

**Pull request merged**